src/spirc.rs

```diff
@@ -43,14 +45,26 @@ struct SpircInternal {
     devices: HashMap<String, String>,
 }
 
+#[derive(Clone)]
+pub struct State {
```

Comment:
see my previous answer about why I removed the update_time, I think when I continue with the changes it will become clear